### PR TITLE
YAML parser used for metadata parsing

### PR DESCRIPTION
**Pull request by grgprarup**
This PR refactors the manual metadata parsing and `js-yaml` parser is used. 
Fixes https://github.com/JankariTech/blog/issues/42

### CODE REVIEW

File: src/assets/TestingWithCypress/cypress_blog_1.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: E2E Testing with Cypress: Introduction
+title: E2E Testing with Cypress - Introduction
```
[the remark this answers is not in the document]
Comment (grgprarup) [reply src/assets/TestingWithCypress/cypress_blog_1.md:2]: Yeah, you are right. That will break the parsing so, I changed the `:` to `-`. Use of `:` metadata should be restricted.

[the remark this answers is not in the document]
Comment (saw-jan) [reply src/assets/TestingWithCypress/cypress_blog_1.md:2]: how will it behave if someone creates a blog PR that has `:` in meta? will the PR be blocked by CI? Do we have/need test for it?

[the remark this answers is not in the document]
Comment (grgprarup) [reply src/assets/TestingWithCypress/cypress_blog_1.md:2]: We have a linter for PR, we can add the rule to that. And also need to add restrictions/instructions in meta-format docs.

[the remark this answers is not in the document]
Comment (saw-jan) [reply src/assets/TestingWithCypress/cypress_blog_1.md:2]: would be great to have tests in this PR and docs as well

File: lint/check.js [hunk 2]
```diff
@@ -68,6 +66,14 @@ log.info("-------------------------------------")
 
 function lintMeta (key) {
   const peekData = extractMeta(markdownFiles[key])
+
+  // check if the metadata is extracted properly
+  if (peekData instanceof Error && peekData.name === "YAMLException") {
+    log.error(`Invalid metadata in ${key}.\nError: ${peekData.message}`)
```
Comment (saw-jan) [inline lint/check.js:72]: Others generally look good.
```suggestion
    log.error(`Invalid metadata in '${key}.md'\nError: ${peekData.message}\n`)
```

**Review by saw-jan — APPROVED** [review]

File: lint/check.js (hunk 2 above)
Comment (grgprarup) [reply lint/check.js]: Thanks for the suggestions. :+1:

**Pull request merged**